cmd/clusterctl/client/repository/template.go

```diff
@@ -30,10 +30,15 @@ import (
 // 1. Checks for all the variables in the cluster template YAML file and replace with corresponding config values
 // 2. Ensure all the cluster objects are deployed in the target namespace.
 type Template interface {
-	// Variables required by the template.
-	// This value is derived by the template YAML.
+	// Variables used by the template.
+	// This value is derived from the template YAML.
 	Variables() []string
```

Comment:
Do we lose the ability this way to retrieve only a list of required variables?

Reply:
The comment was incorrect: `Variables` has always been just a list of the variables we found when parsing the template, not those that are required. 

We didn't change the way `Variables` is constructed, and actually we only know which are required or optional through `VariableMap`.